internal/gatewayapi/translator.go

```diff
@@ -184,3 +200,103 @@ func (t *Translator) GetRelevantGateways(gateways []*v1beta1.Gateway) []*Gateway
 
 	return relevant
 }
+
+func (t *Translator) GetRelevantRoutes(relevantGateways []*GatewayContext,
+	httpRoutes []*v1beta1.HTTPRoute,
+	grpcRoutes []*v1alpha2.GRPCRoute,
+	tlsRoutes []*v1alpha2.TLSRoute,
+	tcpRoutes []*v1alpha2.TCPRoute,
+	udpRoutes []*v1alpha2.UDPRoute) []RouteContext {
+
+	relevantRoutes := []RouteContext{}
+
+	// build a map of the relevant Gateways for faster lookup
+	gateways := map[types.NamespacedName]bool{}
+	for _, gw := range relevantGateways {
+		gw := gw
+		gateways[types.NamespacedName{Name: gw.Name, Namespace: gw.Namespace}] = true
+	}
+
+	// Check all the different types of routes to see if any belong to one of our relevant gateways
+	for _, httpRoute := range httpRoutes {
+		for _, parentGW := range httpRoute.Spec.ParentRefs {
+			key := types.NamespacedName{Name: string(parentGW.Name)}
+			if parentGW.Namespace != nil {
+				key.Namespace = string(*parentGW.Namespace)
```

Comment:
when `parentGW.Namespace == nil` ,  `key.Namespace = httpRoute.Namespace` and the same is true for all other route types